tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -570,6 +572,44 @@ public function testAlterTableScenario()
         }
     }
 
+
+    public function testAlterTableInNamespace()
```

Comment:
I think the functional testing could be reduced to making sure that if the platform supports schemas, then the table names returned by `listTableNames()` are fully qualified (i.e. are formatted as `schema-name.table.name`). The same seems already true for PostgreSQL.

At the same time, for the SQL Server specifically, if a table belongs to the `dbo` schema, it should be returned without the schema prefix for BC.

Does it fairly describe the fix?

Reply:
Yes, that's correct. I'll implement it in that way.